rct229/rulesets/ashrae9012019/section18/section18rule2.py

```diff
@@ -0,0 +1,290 @@
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            data_items={"climate_zone": ("baseline", "weather/climate_zone")},
+        )
+
+    def create_data(self, context, data):
+        rmd_b = context.baseline
+        rmd_p = context.proposed
+        climate_zone_b = data["climate_zone"]
+
+        baseline_system_types_dict_b = get_baseline_system_types(rmd_b)
+        applicable_hvac_sys_ids_b = [
+            hvac_id
+            for sys_type in baseline_system_types_dict_b
+            for target_sys_type in APPLICABLE_SYS_TYPES
+            if baseline_system_type_compare(sys_type, target_sys_type, False)
+            for hvac_id in baseline_system_types_dict_b[sys_type]
+        ]
+
+        zones_and_terminal_unit_list_dict_b = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmd_b)
+        )
+        lab_zone_hvac_systems = get_lab_zone_hvac_systems(
+            rmd_b, rmd_p, climate_zone_b, False
+        )  # TODO the last arg is leap_year
+        lab_zone_exhaust_b = get_building_total_lab_exhaust_from_zone_exhaust_fans(
+            rmd_b
+        )
+
+        # create a hvac data dict
+        hvac_data_b = {}
+        for hvac_id_b in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            rmd_b,
+        ):
+            if hvac_id_b in applicable_hvac_sys_ids_b:
+                hvac_data_b[hvac_id_b] = {
+                    "is_zone_single_zone_sys_b": any(
+                        [
+                            sys_type in APPLICABLE_SYS_TYPES
+                            for sys_type in APPLICABLE_SYS_TYPES
+                        ]
+                    ),
+                    "does_sys_only_serve_lab_b": (
+                        hvac_id_b in lab_zone_hvac_systems["lab_zones_only"]
+                        and lab_zone_exhaust_b > AIRFLOW_15000_CFM
+                        and len(lab_zone_hvac_systems["lab_zones_only"]) == 1
+                    ),
+                    "does_sys_serve_lab_b": (
+                        hvac_id_b in lab_zone_hvac_systems["lab_zones_only"]
+                        and lab_zone_exhaust_b <= AIRFLOW_15000_CFM
+                    ),
+                    "does_sys_serve_lab_and_other_b": (
+                        hvac_id_b in lab_zone_hvac_systems["lab_and_other"]
+                        and lab_zone_exhaust_b <= AIRFLOW_15000_CFM
+                    ),
+                }
+
+                do_multi_zone_evaluation = (
+                    hvac_data_b[hvac_id_b]["is_zone_single_zone_sys_b"]
+                    or not hvac_data_b[hvac_id_b]["does_sys_only_serve_lab_b"]
+                    or not hvac_data_b[hvac_id_b]["does_sys_serve_lab_b"]
+                    or not hvac_data_b[hvac_id_b]["does_sys_serve_lab_and_other_b"]
+                )
```

Comment:
Will need to add the `lab_zone_exhaust_b` in here. 
In RDS, the `do_multi_zone_evaluation` is initialized as a True flag. It turned to False if any one of the conditions met. So it will be `not any([hvac_data_b[hvac_id_b]["does_sys_only_serve_lab_b"], hvac_data_b[hvac_id_b]["does_sys_serve_lab_b"] ....])`

Reply:
Addressed.